File: src/com/suushiemaniac/cubing/alglib/move/modifier/ImageLetterModifier.java
package com.suushiemaniac.cubing.alglib.move.modifier;

public class ImageLetterModifier implements Modifier {
	public static final ImageLetterModifier INST = new ImageLetterModifier();

	@Override
	public String toFormatString() {
		return "";
	}

	@Override
	public Modifier inverse() {
		return this;
	}

	@Override
	public Modifier merge(Modifier other) {
		return null;
	}

	@Override
	public int hashCode() {
		return 0;
	}

	@Override
	public boolean equals(Object obj) {
		return obj instanceof ImageLetterModifier;
	}
}

File: src/main/kotlin/com/suushiemaniac/cubing/alglib/move/modifier/ImageLetterModifier.kt
package com.suushiemaniac.cubing.alglib.move.modifier

class ImageLetterModifier : Modifier {
    override val notation = ""

    override fun inverse() = this
    override fun merge(other: Modifier) = other

    override fun hashCode() = 0
    override fun equals(other: Any?) = other === INST

    companion object {
        val INST = ImageLetterModifier()
    }
}
